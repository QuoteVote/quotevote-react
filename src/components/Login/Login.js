import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { makeStyles } from '@material-ui/core/styles'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import Grid from '@material-ui/core/Grid'
import Link from '@material-ui/core/Link'
import InputAdornment from '@material-ui/core/InputAdornment'

import FaceIcon from '@material-ui/icons/Face'
import LockIcon from '@material-ui/icons/Lock'

import CardBody from '../../mui-pro/Card/CardBody'
import Card from '../../mui-pro/Card/Card'

import 'fontsource-montserrat'

const useStyles = makeStyles({
  header: {
    fontFamily: 'Montserrat',
    fontWeight: 700,
  },
  card: {
    width: 350,
    // height: 459
  },
  loginButton: {
    textTransform: 'none',
  },
  forgotPassword: {
    textTransform: 'none',
  },
  textfield: {
    marginBottom: 20,
  },
  icon: {
    color: '#495057',
  },
  link: {
    color: '#00bcd4',
  },
})

function LoginForm({ onChange, onClick }) {
  const classes = useStyles()

  const handleInputChange = (e) => onChange({ [e.target.name]: e.target.value })

  return (
    <>
      <TextField
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <FaceIcon className={classes.icon} />
            </InputAdornment>
          ),
        }}
        className={classes.textfield}
        placeholder="Username"
        fullWidth
        name="username"
        id="username"
        onChange={handleInputChange}
      />
      <TextField
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <LockIcon className={classes.icon} />
            </InputAdornment>
          ),
        }}
        className={classes.textfield}
        placeholder="Password"
        fullWidth
        name="password"
        id="password"
        type="password"
        onChange={handleInputChange}
      />
      <Button
        className={classes.loginButton}
        size="large"
        color="primary"
        variant="contained"
        fullWidth
        onClick={onClick}
      >
        <Typography variant="body1" color="secondary">
          Log in
        </Typography>
      </Button>
    </>
  )
}

LoginForm.propTypes = {
  onChange: PropTypes.func,
  onClick: PropTypes.func,
}

function Login({ login = () => {} }) {
  const [password, setPassword] = useState('')
  const [username, setUsername] = useState('')

  const classes = useStyles()

  const handleInputChange = (val) => {
    setUsername(val.username)
    setPassword(val.password)
  }

  const handleLoginButtonClick = () => {
    login(username, password)
  }

  return (
    <Card className={classes.card}>
      <CardBody>
        <Grid
          container
          direction="column"
          alignItems="center"
          justify="space-evenly"
          spacing={2}
        >
          <Grid item>
            <Typography variant="h6">Login</Typography>
          </Grid>
          <Grid item>
            <LoginForm
              onChange={handleInputChange}
              onClick={handleLoginButtonClick}
            />
          </Grid>
          <Grid item>
            <Typography variant="body1">
              <Link className={classes.link} href="/forgot">
                Forgot password?
              </Link>
            </Typography>
          </Grid>
          <Grid item>
            <Typography variant="body1">
              No account?
              <span />
              <Link className={classes.link} href="/request-access">
                Request Access
              </Link>
            </Typography>
          </Grid>
        </Grid>
      </CardBody>
    </Card>
  )
}

Login.propTypes = {
  login: PropTypes.func,
}

export default Login
