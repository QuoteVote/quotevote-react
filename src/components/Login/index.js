import React from 'react'
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import { makeStyles } from '@material-ui/core/styles'
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import GridContainer from 'mui-pro/Grid/GridContainer'
import GridItem from 'mui-pro/Grid/GridItem'
import Typography from '@material-ui/core/Typography'
import CustomInput from 'mui-pro/CustomInput/CustomInput'
import Face from '@material-ui/icons/Face'
import { InputAdornment } from '@material-ui/core'
import styles from 'assets/jss/material-dashboard-pro-react/views/loginPageStyle'


const useStyles = makeStyles(styles)

const Login = (props) => {
  const { open, setOpen } = props
  const classes = useStyles()
  console.log('open: ', open);
  return (
    <Dialog open={open} onClose={() => setOpen(false)} aria-labelledby="form-dialog-title" maxWidth="xs">
      <DialogContent>
        <GridContainer>
          <GridItem xs={12}>Log in with Google</GridItem>
          <GridItem xs={12}>Log in with Facebook</GridItem>
          <GridItem xs={12}>Log in with Twitter</GridItem>
          <GridItem xs={12}>
            <Typography align="center" variant="subtitle1" style={{ textTransform: "none" }}>or</Typography>
          </GridItem>
          <GridItem xs={12}>
            <CustomInput
              labelText="Email..."
              id="username"
              formControlProps={{
                fullWidth: true,
              }}
              inputProps={{
                startAdornment: (
                  <InputAdornment position="end">
                    <Face className={classes.inputAdornmentIcon} />
                  </InputAdornment>
                ),
                // onChange: (e) => handleInputs(e),
              }}
              // error={loginError}
            />
          </GridItem>
          <GridItem xs={12}>Password</GridItem>
          <GridItem xs={12}>Log in</GridItem>
          <GridItem xs={12}>Forgot Password</GridItem>
          <GridItem xs={12}>No account? Request invite</GridItem>
        </GridContainer>
      </DialogContent>
    </Dialog>
  )
}

export default Login