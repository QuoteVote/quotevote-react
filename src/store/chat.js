/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'

const chat = createSlice({
  name: 'chat',
  initialState: {
    submitting: false,
    selectedRoom: null,
  },
  reducers: {
    CHAT_SUBMITTING: (state, action) => {
      state.submitting = action.payload
    },
    SELECTED_CHAT_ROOM: (state, action) => {
      state.selectedRoom = action.payload
    },
  },
})

export const { CHAT_SUBMITTING, SELECTED_CHAT_ROOM } = chat.actions

export default chat.reducer
