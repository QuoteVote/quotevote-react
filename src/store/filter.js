/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'

const uiSlice = createSlice({
  name: 'filter',
  initialState: {
    filter: {
      visibility: false,
      value: ['POSTED'],
    },
    date: {
      visibility: false,
      value: '',
    },
    search: {
      visibility: false,
      value: '',
    },
  },
  reducers: {
    FILTER_VISIBILITY: (state, action) => {
      state.filter.visibility = action.payload
    },
    FILTER_VALUE: (state, action) => {
      state.filter.value = action.payload
    },
    DATE_VISIBILITY: (state, action) => {
      state.date.visibility = action.payload
    },
    DATE_VALUE: (state, action) => {
      state.date.value = action.payload
    },
    SEARCH_VISIBILITY: (state, action) => {
      state.search.visibility = action.payload
    },
    SEARCH_VALUE: (state, action) => {
      state.search.value = action.payload
    },
  },
})

export const {
  FILTER_VISIBILITY,
  FILTER_VALUE,
  DATE_VISIBILITY,
  DATE_VALUE,
  SEARCH_VISIBILITY,
  SEARCH_VALUE,
} = uiSlice.actions

export default uiSlice.reducer
