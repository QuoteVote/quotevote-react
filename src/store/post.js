/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit'

const postSlice = createSlice({
  name: 'post',
  initialState: {
    selectedPost: {
      id: null,
    },
  },
  reducers: {
    SET_SELECTED_POST: (state, action) => {
      state.selectedPost.id = action.payload
    },
  },
})

export const { SET_SELECTED_POST } = postSlice.actions

export default postSlice.reducer
