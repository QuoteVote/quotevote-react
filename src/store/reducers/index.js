export { default as postReducer } from './postReducer'
export { default as loginReducer } from './loginReducer'
export { default as chatReducer } from './chatReducer'
export { default as appReducer } from './appReducer'
