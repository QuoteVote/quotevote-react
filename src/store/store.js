import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import postReducer from 'store/post'
import userReducer from 'store/user'

const rootReducer = combineReducers({
  user: userReducer,
  post: postReducer,
})

// Add middleware the end of the array
const middleware = [...getDefaultMiddleware()]

const devMode = process.env.NODE_ENV === 'development'

export default configureStore({
  reducer: rootReducer,
  middleware,
  devTools: devMode,
})
