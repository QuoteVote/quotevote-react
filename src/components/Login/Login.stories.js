// Important stuff, must always be imported on a storybook file
import React from 'react'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withA11y } from '@storybook/addon-a11y'

// If you want to apply theme
import { MuiThemeProvider as ThemeProvider } from '@material-ui/core/styles'

// The component
import Login from './Login'

import theme from '../../themes/MainTheme'

// Story config
export default {
  title: 'Login',
  argTypes: { onClick: { action: 'clicked' }, onChange: { action: 'typing' } },
  component: Login,
  decorators: [withKnobs, withA11y],
}

function Wrapper() {
  return (
    <ThemeProvider theme={theme}>
      <Login />
    </ThemeProvider>
  )
}

export const base = () => <Wrapper />

base.story = {
  parameters: {
    jest: ['Login.test.js'],
  },
}
