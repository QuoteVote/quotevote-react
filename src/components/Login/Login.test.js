import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from 'store/store'

// Component being tested
import Login from './Login'

describe('Login test -', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Login />
      </Provider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it.todo('validate email')
  it.todo('validate password')
  it.todo('submits upon button press')
})
