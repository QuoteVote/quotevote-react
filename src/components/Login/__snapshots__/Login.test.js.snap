// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test - renders correctly 1`] = `
<div
  class="makeStyles-card-8 makeStyles-card-2"
>
  <div
    class="makeStyles-cardBody-25"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <p
          class="MuiTypography-root makeStyles-header-1 MuiTypography-body1"
        >
          Login
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-textfield-5 MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 11.75c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zm6 0c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37C11.07 8.33 14.05 10 17.42 10c.78 0 1.53-.09 2.25-.26.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                id="username"
                name="username"
                placeholder="Email/Username"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-textfield-5 MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-loginButton-3 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Log in
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-7 MuiTypography-colorPrimary"
            href="/auth/forgot"
          >
            Forgot password?
          </a>
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          No account?
          <span
            style="margin-right: 5px;"
          />
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-7 MuiTypography-colorPrimary"
          href="/auth/request-access"
        >
          Request Access
        </a>
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-7 MuiTypography-colorPrimary"
          href="https://github.com/QuoteVote/quotevote-react"
          target="_blank"
          rel="noopener noreferrer"
        >
          Source Code
        </a>
      </p>
    </div>
  </div>
  </div>
</div>
`;
